From 42e55087ed4360d47e56be100460b1ba653c0eb6 Mon Sep 17 00:00:00 2001
Message-Id: <42e55087ed4360d47e56be100460b1ba653c0eb6.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Fri, 26 Sep 2014 15:50:52 +0530
Subject: [PATCH 12/75] MUST_REBASE [VPG]: drm/i915/adf: Disable DPST in i915
 if adf config is enabled.

Will be enabled in ADF driver once backlight support is there. Once DPST
is succesfully running in ADF, it will be removed from i915

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I461330d0833ea5e03aa57514dcc72a51dfeece71
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index ac0789f..10796a7 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -2289,7 +2289,8 @@ enum context_submission_status {
 #define HAS_FW_BLC(dev) (INTEL_INFO(dev)->gen > 2)
 #define HAS_PIPE_CXSR(dev) (INTEL_INFO(dev)->has_pipe_cxsr)
 #define HAS_FBC(dev) (INTEL_INFO(dev)->has_fbc)
-#define I915_HAS_DPST(dev) (INTEL_INFO(dev)->has_dpst)
+#define I915_HAS_DPST(dev) (INTEL_INFO(dev)->has_dpst && \
+			    !(i915.enable_intel_adf))
 
 #define HAS_IPS(dev)		(IS_ULT(dev) || IS_BROADWELL(dev))
 
-- 
1.7.9.5

